sql/report_queries/loans_and_renewals_counts/loans_and_renewals_counts.sql

```diff
@@ -1,175 +1,168 @@
-    WHERE
-		itpl."name" = (SELECT items_permanent_location_filter FROM parameters)
-		OR ittl."name" = (SELECT items_temporary_location_filter FROM parameters)
-		OR itel."name" = (SELECT items_effective_location_filter FROM parameters)
-		OR lib."name" = (SELECT library_filter FROM parameters)
-		OR cmp."name" = (SELECT campus_filter FROM parameters)
-		OR ins."name" = (SELECT institution_filter FROM parameters)
-),
-loan_count AS (
-	SELECT 
-		id, 
-		COUNT(id) AS num_loans,
-		SUM(renewal_count) AS num_renewals
-	FROM circulation_loans
-	GROUP BY id
-),
-loan_details AS (
-	SELECT 
-		l.loan_date :: DATE,
-		l.due_date AS loan_due_date,
-		l.return_date AS loan_return_date,
-		l.item_status AS loan_status,
-		l.item_id,
-		l.id AS loan_id,
-		lc.num_loans,
-		lc.num_renewals,
-		l.patron_group_id_at_checkout,
-		i.material_type_id,
-		mt.name AS material_type_name,
-		g.group AS patron_group_name,
-		lp.name AS loan_policy_name,
-		plt.name AS permanent_loan_type_name,
-		tlt.name AS temporary_loan_type_name,
-		location_filtering.temp_location_name,
-		location_filtering.perm_location_name,
-		location_filtering.effective_location_name,
-		location_filtering.lib_name,
-		location_filtering.campus_name,
-		location_filtering.institute_name
-	FROM circulation_loans as l
-	LEFT JOIN inventory_items AS i
-        ON l.loan_policy_id=i.id
-	LEFT JOIN circulation_loan_policies as lp
-	    ON l.loan_policy_id=lp.id
-	LEFT JOIN inventory_loan_types as plt
-	    ON i.permanent_loan_type_id=plt.id
-	LEFT JOIN inventory_loan_types as tlt
-        ON i.temporary_loan_type_id=tlt.id
-	LEFT JOIN inventory_material_types as mt
-        ON i.material_type_id=mt.id
-	LEFT JOIN user_groups as g
-        ON l.patron_group_id_at_checkout=g.id
-	LEFT JOIN loan_count AS lc
-        ON l.id=lc.id
-	LEFT JOIN location_filtering 
-        ON l.item_id= location_filtering.item_id)
---MAIN QUERY
-SELECT 
-	(SELECT start_date :: VARCHAR FROM parameters) ||
-     ' to ' :: VARCHAR ||
-    (SELECT end_date :: VARCHAR FROM parameters) AS date_range,
-		loan_due_date,
-		loan_return_date,
-		loan_status,
-		num_loans,
-		num_renewals,
-		patron_group_name,
-		material_type_name,
-		loan_policy_name,
-		permanent_loan_type_name,
-		temporary_loan_type_name,
-		loan_date,
-		temp_location_name,
-		perm_location_name,
-		effective_location_name,
-		lib_name,
-		campus_name,
-		institute_name
-	FROM loan_details
-WHERE loan_date >= (SELECT start_date FROM parameters)
-AND loan_date < (SELECT end_date FROM parameters)
-;
+        (
+            SELECT
+                start_date::varchar
+            FROM
+                parameters) || ' to '::varchar || (
+        SELECT
+            end_date::varchar
+        FROM
+            parameters) AS date_range,
+    li.loan_date::date,
+    li.loan_due_date AS loan_due_date,
+    li.loan_return_date AS loan_return_date,
+    li.loan_status AS loan_status,
+    '1'::int AS num_loans, -- each row is a single loan
```

Comment:
I wondered if this column is useful.  I would think that `sum(num_loans)` is the same as `count(*)`.  I can see it being more user friendly since `sum(num_loans)` would not require creating a new column; or confusing because it does not convey information in the context of this query.

Reply:
I'm imagining Excel users who will not be using database tools to compute this sum. If they know how to use pivot tables, they won't need the count. Otherwise, they may prefer a number they can sum rather than using Excel's COUNT function, which I think may be less intuitive and which can produce problematic results if you use it on a column with blanks.